### PR TITLE
chore: update arch docs and add mermaid

### PR DESCRIPTION
**Pull request by rokroskar**
Replaces outdated graphviz architecture diagram with a mermaid-based one. Installs & configures mermaid.

addresses #2565

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "sphinx.ext.todo",
     "sphinx.ext.viewcode",
     "sphinxcontrib.spelling",
+    'sphinxcontrib.mermaid',
```
Comment (lorenzo-cavazzi) [inline docs/conf.py:45]:
```suggestion
    "sphinxcontrib.mermaid",
```

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
Oh, I guess it's time for a review... Sorry, this slipped out of my radar :sweat_smile: 

Screenshot_20220607_141303

Looks great! And the automatically generated graph has surprisingly no arrows overlapping. This won't happen again, we should never ever change it!

The one detail I would modify is the arrow from KG to GitLab. KG is actually the only backend service that gets server-side events from GitLab through the webhook. Not sure if you want to make it explicit here.

Apart from that, just a tiny detail to update on the conf file.

**Comment by rokroskar** [comment]
@lorenzo-cavazzi I made the arrow between gitlab and kg be bidirectional - if I add a second arrow going in the other direction it completely explodes the neat graph... that goes agains

> we should never ever change it!

**Review by lorenzo-cavazzi — APPROVED** [review]
Lgtm!

**Pull request merged**